src/material-experimental/mdc-slide-toggle/_slide-toggle-theme.scss

```diff
@@ -5,25 +5,22 @@
 @import '@material/theme/functions.import';
 @import '../mdc-helpers/mdc-helpers';
 
+// Apply the theme colors with the MDC color-specific slide-toggle mixins.
+@mixin _apply-mdc-switch-color($color) {
+  .mdc-switch {
+    @include mdc-switch-toggled-on-track-color($color);
+    @include mdc-switch-toggled-on-thumb-color($color);
+    @include mdc-switch-toggled-off-track-color(mdc-theme-prop-value(on-surface));
+    @include mdc-switch-toggled-off-thumb-color(mdc-theme-prop-value(surface));
```

Comment:
Isn't this a bit more fragile since MDC could add another mixin to `mdc-switch-without-ripple`?

Reply:
I think I could see it either way - the variables seem fragile to me since they could simply create more and we wouldn't know about that either. This approach feels more like using an open API rather than overriding the internals. 

Of course this may all just be moot since they want to reconsider how colors are provided anyways.